### PR TITLE
Make code and reason WebTransportCloseInfo non-optional

### LINKED ISSUE #366: Session closure without close info
1. When a client close a WebTransport session __without__ providing WebTransportCloseInfo, Do we want to send a CLOSE_WEBTRANSPORT_SESSION capsule (with the default value)? Or do we close the CONNECT stream without sending a CLOSE_WEBTRANSPORT_SESSION capsule?
 1. When a server closes the CONNECT stream without sending a CLOSE_WEBTRANSPORT_SESSION capsule, do we want to resolve the `.closed` promise with an empty object, or do we want to resolve it with `{closeCode: 0, reason: ""}`?

### PR DESCRIPTION
**Pull request by yutakahirano**
Fixes #366.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webtransport/pull/379.html" title="Last updated on Dec 8, 2021, 12:15 AM UTC (4dd63c7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webtransport/379/3ddc1a8...4dd63c7.html" title="Last updated on Dec 8, 2021, 12:15 AM UTC (4dd63c7)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -847,16 +847,12 @@ these steps.
        1. [=Cleanup=] |transport| with |error|, |error| and true.
        1. Abort these steps.
      1. Let |session| be |transport|'s [=[[Session]]=].
+     1. Let |code| be |closeInfo|.{{WebTransportCloseInfo/closeCode}}.
      1. Let |reason| be an empty [=byte sequence=].
```
Comment (jan-ivar) [inline index.bs:851]: Github won't let me make suggestions across deleted lines apparently, so doing this in two steps.
```suggestion
```

File: index.bs [hunk 2]
```diff
@@ -847,16 +847,12 @@ these steps.
...
-     1. If |closeInfo|.{{WebTransportCloseInfo/reason}} exists, set |reason| be
-        |closeInfo|.{{WebTransportCloseInfo/reason}}, [=UTF-8 encoded=].
+     1. Set |reason| to |closeInfo|.{{WebTransportCloseInfo/reason}}, [=UTF-8 encoded=].
      1. If |reason|'s [=byte sequence/length=] exceeds 1024, then set |reason| to a
         [=byte sequence/prefix=] of |reason| whose length is 1024.
```
Comment (jan-ivar) [inline index.bs:854]: (Side-observation from looking at the wpt test) Doesn't this risk clipping a multi-byte UTF8 sequence?

**Review by jan-ivar — APPROVED** [review]

File: index.bs (hunk 2 above)
Comment (martinthomson) [reply index.bs:853]: Issue?

Comment (jan-ivar) [reply index.bs:853]: Filed as https://github.com/w3c/webtransport/issues/382

**Pull request merged**